lib/bitcoin/lock/LockResolver.ts

```diff
@@ -0,0 +1,139 @@
+    }
+
+    // Now that the lock identifier has been verified, return the lock information
+    const serializedLockIdentifier = LockIdentifierSerializer.serialize(lockIdentifier);
+
+    return {
+      identifier: serializedLockIdentifier,
+      amountLocked: lockTransaction.outputs[0].satoshis,
+      lockEndTransactionTime: lockUntilBlock,
+      linkedWalletAddress: walletAddressObj.toString()
+    };
+  }
+
+  /**
+   * Checks whether the redeem script is indeed a lock script.
+   * @param redeemScript The script to check.
+   * @returns A touple where [0] == true if the script is a lock script; false otherwise, [1] == if [0] is true then the lockUntilBlock.
+   */
+  private static isRedeemScriptALockScript (redeemScript: Script): [boolean, number] {
+
+    // Split the script into parts and verify each part
+    const scriptAsmParts = redeemScript.toASM().split(' ');
+
+    // Verify different parts
+    const isScriptValid =
+      scriptAsmParts.length === 8 &&
+      scriptAsmParts[1] === 'OP_NOP2' &&
+      scriptAsmParts[2] === 'OP_DROP' &&
+      scriptAsmParts[3] === 'OP_DUP' &&
+      scriptAsmParts[4] === 'OP_HASH160' &&
+      scriptAsmParts[6] === 'OP_EQUALVERIFY' &&
+      scriptAsmParts[7] === 'OP_CHECKSIG';
+
+    if (isScriptValid) {
+      const lockUntilBlockBuffer = Buffer.from(scriptAsmParts[0], 'hex');
+      const lockUntilBlock = lockUntilBlockBuffer.readIntLE(0, lockUntilBlockBuffer.length);
+
+      return [isScriptValid, lockUntilBlock];
+    }
+
+    return [isScriptValid, 0];
+  }
+
+  /**
+   * Checks whether the specified output is a "paytoscript" type output to the specified script.
+   * @param bitcoinOutput The freeze output from the bitcoin transaction.
+   * @param targetScript The expected redeem script.
+   */
+  private static isOutputPayingToTargetScript (bitcoinOutput: BitcoinOutputModel, targetScript: Script): boolean {
+    const targetScriptHashOut = Script.buildScriptHashOut(targetScript);
+
+    return bitcoinOutput.scriptAsmAsString === targetScriptHashOut.toASM();
+  }
+
+  /**
+   * Checks whether the specified redeem script is paying to the target wallet address.
+   * @param redeemScript The redeem script representing a lock script.
+   * @param targetWalletAddress The wallet address to be verified.
+   */
+  private static isRedeemScriptPayingToTargetWallet (redeemScript: Script, targetWalletAddress: Address): boolean {
+    // Convert the wallet address into the standard "publickeyhash" bitcoin output
+    const publicKeyHashOutput = Script.buildPublicKeyHashOut(targetWalletAddress);
+    const publicKeyHashOutputAsHex = publicKeyHashOutput.toHex();
+
+    // If the above output is the suffix in the redeem script then it is paying to the target wallet
+    return redeemScript.toHex().endsWith(publicKeyHashOutputAsHex);
+  }
+
+  private static createScriptFromHexInput (redeemScriptAsHex: string): Script {

```

Comment:
Nit: "Input" sounds like transaction input.

```suggestion
  private static createScript (redeemScriptAsHex: string): Script {

```